examples/download_and_show_image.rb

```diff
@@ -0,0 +1,32 @@
+Shoes.app do
+  stack do
+    title "Downloading Google image", size: 16
+    @status = para "Enter URL and press Enter:"
+    @download_completed = false
+
+    url_edit = edit_line
+    caption "here is a cute cat url: https://img.freepik.com/premium-photo/funny-smart-cat-professor-with-glasses-illustration-generative-ai_845977-709.jpg?w=700&h=300"
+
+    button 'Download Image!',color:"#3d5a80",padding_bottom:"8",padding_top:"8",text_color:"white",font_size:"16" do
+      download(url_edit.text, save: "downloaded_image.png") do
+        @status.replace "Download completed."
+        @download_completed = true
+      end
+    end
+
+    Thread.new do
+      loop do
```

Comment:
I feel funny about having the background thread in an example here, because that's *not* always okay with various display libraries. I think we want to try to keep (new) examples as squeaky-clean as possible.

Reply:
ah oh sorry i will change

Reply:
changed